src/main/java/ee/tuleva/onboarding/auth/session/GenericSessionStore.java

```diff
@@ -1,49 +1,57 @@
-import org.springframework.web.context.request.ServletRequestAttributes;
 
 @Component
+@RequiredArgsConstructor
 public class GenericSessionStore {
 
-  private static final String GENERIC_SESSION_STORE_ATTRIBUTES =
-      GenericSessionStore.class.getName() + ".attributes";
+  private final SessionAttributeRepository repository;
+  private final AuthenticationHolder authenticationHolder;
 
   public <T extends Serializable> void save(T sessionAttribute) {
-    getSessionAttributes().put(sessionAttribute.getClass().getName(), sessionAttribute);
-  }
+    Long userId = authenticationHolder.getAuthenticatedPerson().getUserId();
+    String attributeName = sessionAttribute.getClass().getName();
+    byte[] attributeBytes;
 
-  public <T extends Serializable> Optional<T> get(Class<?> clazz) {
-    @SuppressWarnings("unchecked")
-    T sessionAttribute = (T) getSessionAttributes().get(clazz.getName());
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        ObjectOutput out = new ObjectOutputStream(bos)) {
 
-    if (sessionAttribute == null) {
-      return Optional.empty();
+      out.writeObject(sessionAttribute);
+      attributeBytes = bos.toByteArray();
+    } catch (IOException e) {
+      throw new RuntimeException("Error serializing session attribute", e);
     }
 
-    return Optional.of(sessionAttribute);
-  }
+    repository.findByUserIdAndAttributeName(userId, attributeName).ifPresent(repository::delete);
 
-  private static <T extends Serializable> Map<String, T> getSessionAttributes() {
-    @SuppressWarnings("unchecked")
-    Map<String, T> attributes =
-        (Map<String, T>) getSession().getAttribute(GENERIC_SESSION_STORE_ATTRIBUTES);
-    if (attributes == null) {
-      attributes = new HashMap<>();
-      getSession().setAttribute(GENERIC_SESSION_STORE_ATTRIBUTES, attributes);
-    }
-    return attributes;
+    SessionAttribute newAttribute =
+        SessionAttribute.builder()
+            .userId(userId)
+            .attributeName(attributeName)
+            .attributeValue(attributeBytes)
+            .build();
+    repository.save(newAttribute);
   }
 
-  private static HttpSession getSession() {
-    ServletRequestAttributes attr =
-        (ServletRequestAttributes) RequestContextHolder.currentRequestAttributes();
-    return attr.getRequest().getSession(true);
+  public <T extends Serializable> Optional<T> get(Class<T> clazz) {
+    Long userId = authenticationHolder.getAuthenticatedPerson().getUserId();
+    return repository
+        .findByUserIdAndAttributeName(userId, clazz.getName())
+        .flatMap(
+            attribute -> {
+              try (ObjectInput in =
+                  new ObjectInputStream(new ByteArrayInputStream(attribute.getAttributeValue()))) {
+                @SuppressWarnings("unchecked")
+                T result = (T) in.readObject();
+                return Optional.of(result);
+              } catch (IOException | ClassNotFoundException e) {
```

Comment:
We should never catch `ClassNotFoundException`!! This is a classpath configuration error that should be resolved